generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FunctionDataFetcher.kt

```diff
@@ -125,24 +128,36 @@ open class FunctionDataFetcher(
         argumentName: String,
         argumentValue: Any?
     ): Any? = when {
-        param.isList() -> {
-            val argumentClass = param.type.getTypeOfFirstArgument().getJavaClass()
-            val jacksonCollectionType = objectMapper.typeFactory.constructCollectionType(List::class.java, argumentClass)
-            objectMapper.convertValue(argumentValue, jacksonCollectionType)
-        }
         param.type.isOptionalInputType() -> {
             when {
                 !environment.containsArgument(argumentName) -> OptionalInput.Undefined
                 argumentValue == null -> OptionalInput.Defined(null)
                 else -> {
-                    val argumentClass = param.type.getTypeOfFirstArgument().getJavaClass()
-                    val value = objectMapper.convertValue(argumentValue, argumentClass)
+                    val paramType = param.type.getTypeOfFirstArgument()
```

Comment:
When the type is `OptionalInput`, get the info of the wrapped type and convert it as usual using the new `convertValue`